File: src/rag/AdminRAG.ts
/**
 * AdminRAG - RAG module for Admin application queries.
 *
 * Retrieves information exclusively from the data/admin/ directory.
 * Handles: platform administration, internal documentation, monitoring, workflows.
 */

import * as path from "path";
import { BaseRAGModule, RAGResponse } from "./BaseRAGModule";
import { LocalDocumentStore } from "./LocalDocumentStore";

export class AdminRAG implements BaseRAGModule {
  readonly moduleName = "admin_rag";
  private readonly store: LocalDocumentStore;

  constructor(dataRootDir: string) {
    const adminDataDir = path.join(dataRootDir, "admin");
    this.store = new LocalDocumentStore(adminDataDir);
    console.log(`[AdminRAG] Initialized with data from: ${adminDataDir}`);
  }

  async retrieve(query: string, topK: number = 3): Promise<RAGResponse> {
    const results = this.store.search(query, topK);

    const context = results.length > 0
      ? results
          .map(
            (r, i) =>
              `[Source: ${r.source} | Relevance: ${r.score}]\n${r.content}`
          )
          .join("\n\n")
      : "No relevant information found in the admin knowledge base.";

    return {
      context,
      results,
      ragModule: this.moduleName,
    };
  }
}

File: src_python/rag/admin_rag.py
"""
AdminRAG - RAG module for Admin application queries.

Retrieves information exclusively from the data/admin/ directory.
Handles: platform administration, internal documentation, monitoring, workflows.
"""

from __future__ import annotations

import os

from .base_rag_module import BaseRAGModule, RAGResponse
from .local_document_store import LocalDocumentStore


class AdminRAG(BaseRAGModule):
    def __init__(self, data_root_dir: str):
        admin_data_dir = os.path.join(data_root_dir, "admin")
        self._store = LocalDocumentStore(admin_data_dir)
        print(f"[AdminRAG] Initialized with data from: {admin_data_dir}")

    @property
    def module_name(self) -> str:
        return "admin_rag"

    async def retrieve(self, query: str, top_k: int = 3) -> RAGResponse:
        results = self._store.search(query, top_k)

        if results:
            context = "\n\n".join(
                f"[Source: {r.source} | Relevance: {r.score}]\n{r.content}"
                for r in results
            )
        else:
            context = "No relevant information found in the admin knowledge base."

        return RAGResponse(
            context=context,
            results=results,
            rag_module=self.module_name,
        )
